test/unit/controllers/PeerSelectControllerSpec.js

```diff
@@ -29,229 +27,227 @@ describe('PeerSelectController', function () {
+    this.createController()
 
     setTimeout(function () {
-      expect($scope.shareLink.loading).toBe(true)
-      expect($scope.shareLink.url).not.toBeDefined()
+      expect(this.$scope.shareLink.loading).toBe(true)
+      expect(this.$scope.shareLink.url).not.toBeDefined()
       setTimeout(function () {
-        expect($scope.shareLink.url).toBe(expected)
-      }, timeoutTime)
-    }, timeoutTime)
+        expect(this.$scope.shareLink.url).toBe(expected)
+      }, waitOneSecond)
+    }, waitOneSecond)
     done()
   })
 
   it('compiles with a shareLinkPromise that doesn\'t resolve', function (done) {
-    $scope.shareLinkPromise = {
+    var waitOneSecond = this.waitOneSecond
+    this.$scope.shareLinkPromise = {
       then: function (resolve, reject) {
-        setTimeout(reject(), timeoutTime)
+        setTimeout(reject(), waitOneSecond)
       }
     }
-    createController()
+    this.createController()
 
     setTimeout(function () {
-      expect($scope.shareLink.loading).toBe(true)
+      expect(this.$scope.shareLink.loading).toBe(true)
       setTimeout(function () {
-        expect($scope.shareLink).not.toBeDefined()
-      }, timeoutTime)
-    }, timeoutTime)
-    done()
-  })
-
-  it('can select and submit a single dialog without confirmed type', function (done) {
-    createController()
-
-    $scope.dialogSelect('dialogX')
-
-    expect($promiseData).not.toBeDefined()
-    expect($promiseFlag).toBe(true)
-
+        expect(this.$scope.shareLink).not.toBeDefined()
+      }, waitOneSecond)
+    }, waitOneSecond)
     done()
   })
 
-  it('can select and submit a single dialog with confirmed type', function (done) {
-    createController()
-
-    $scope.confirm_type = 'INVITE_TO_GROUP'
-    $scope.dialogSelect('dialogX')
-
-    var expected = {
-      type: 'INVITE_TO_GROUP',
-      peer_id: 'X',
-      peer_data: 'Xpeer'
-    }
-
-    expect($promiseData).toEqual(expected)
-    expect($promiseFlag).toBe(true)
-
-    done()
-  })
+  describe('after initialisation', function () {
+    beforeEach(function () {
+      this.createController()
+    })
 
-  it('can select a dialog', function (done) {
-    createController()
+    it('can select and submit a single dialog without confirmed type', function (done) {
+      this.$scope.dialogSelect('dialogX')
 
-    $scope.multiSelect = true
-    $scope.dialogSelect('dialogX')
+      expect(this.ErrorService.$promiseData).toEqual({})
+      expect(this.promise.$promiseFlag).toBe(true)
 
-    var expected = ['X']
+      done()
+    })
 
-    expect($scope.selectedPeers['X']).toBe('Xpeer')
-    expect($scope.selectedCount).toBe(1)
-    expect($scope.selectedPeerIDs).toEqual(expected)
+    it('can select and submit a single dialog with confirmed type', function (done) {
+      this.$scope.confirm_type = 'INVITE_TO_GROUP'
+      this.$scope.dialogSelect('dialogX')
 
-    done()
-  })
+      var expected = {
+        type: 'INVITE_TO_GROUP',
+        peer_id: 'X',
+        peer_data: 'Xpeer'
```

Comment:
Is `peer_data` related to `peer_id`? If so, you could indicate by string concatenation or computing afer initialisation of the object...